Recent kernels do not update the access time on read.
So do that manually.
See https://github.com/google/jax/issues/7944

Author: Alexander Grund (TU Dresden)

diff --git a/jax/experimental/compilation_cache/file_system_cache.py b/jax/experimental/compilation_cache/file_system_cache.py
index b85969de..4c22d8e3 100644
--- a/jax/experimental/compilation_cache/file_system_cache.py
+++ b/jax/experimental/compilation_cache/file_system_cache.py
@@ -33,6 +33,7 @@ class FileSystemCache(CacheInterface):
     path_to_key = os.path.join(self._path, key)
     if os.path.exists(path_to_key):
       with open(path_to_key, "rb") as file:
+        os.utime(file.fileno())
         return file.read()
     else:
       return None
